mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/jobmaster/WorkerMetricHandler.java

```diff
@@ -280,6 +307,28 @@ public void call() {
                                                     gauges.get(ON_NEXT_GAUGE) / 6.0, numWorkers, ""));
                                 }
                             }
+
+                            double sourceJobDrops = 0;
+                            boolean hasSourceJobDropsMetric = false;
+                            Map<String, String> sourceMetricsRecent = sourceJobMetricsRecent.asMap();
+                            for (Map.Entry<String, WorkerMetrics> worker : sourceJobWorkersMap.entrySet()) {
+                                Map<String, GaugeData> metricGroups = metricAggregator.getAggregates(worker.getValue().getGaugesByMetricGrp());
+                                for (Map.Entry<String, GaugeData> group : metricGroups.entrySet()) {
+                                    String metricKey = worker.getKey() + ":" + group.getKey();
+                                    Map<String, Double> gauges = group.getValue().getGauges();
+                                    if (sourceMetricsRecent.containsKey(metricKey) && gauges.containsKey(DROPPED_COUNTER_METRIC_NAME)) {
+                                        sourceJobDrops += gauges.get(DROPPED_COUNTER_METRIC_NAME);
+                                        hasSourceJobDropsMetric = true;
+                                    }
+                                }
+                            }
+                            if (hasSourceJobDropsMetric) {
```

Comment:
Is there a way to weed out outliers? As in say source job is dropping data while writing to to only 1 worker, then we don't necessarily want to scale up this job right

Reply:
Yes, that's planned in a followup PR. The outlier detector should automatically resubmit the worker when it detects drops are only to 1 worker.